examples/advanced/quickxplain.py

```diff
@@ -0,0 +1,83 @@
+"""
+    CPMpy implementation of the QuickXplain algorithm by Junker.
+        Junker, Ulrich. "Preferred explanations and relaxations for over-constrained problems." AAAI-2004. 2004.
+"""
+
+import cpmpy as cp
+from cpmpy.transformations.get_variables import get_variables
+from cpmpy.transformations.normalize import toplevel_list
+
+
+def quickXplain(soft, hard=[], order=lambda x : 0, solver="ortools"):
+    """
+        Find a preferred minimal unsatisfiable subset of constraints
+        A partial order is imposed on the constraints using the `order` argument
+        Constraints with lower order are preferred to ones with higher order
+
+        :param: soft: soft constraints, list of expressions
+        :param: hard: hard constraints, optional, list of expressions
+        :param: order: optional function to compute order of constraints
+        :param: solver: name of a solver, see SolverLookup.solvernames()
+            "z3" and "gurobi" are incremental, "ortools" restarts the solver
+    """
+
+    soft = toplevel_list(soft, merge_and=False)
+    assump = cp.boolvar(shape=len(soft))
+    solver = cp.SolverLookup.get(solver, cp.Model(hard))
+    solver += assump.implies(soft)
+
+    assert solver.solve(assumptions=assump) is False, "The model should be UNSAT!"
+
+    dmap = dict(zip(assump, soft))
+    core = recurse_explain(list(assump),[],[],order, dmap, solver=solver)
```

Comment:
Useless remark that can be ignored:
Since it does not change anything to the execution, but to be as close as possible to the original algorithm, delta should be = list(assump)
```
core = recurse_explain(list(assump), [], list(assump), order, dmap, solver=solver)
```

Reply:
Hmm not sure about this one, I like the current version as it gives a clear view on the initialization of the recursion.
But we might add a comment saying why we do not put the hard constraints here (we added them to the solver).

About the order of arguments, it really depends on which paper you read and I would like to keep it close to the interface of the `mus` tool

Reply:
Sounds reasonable to me :-)